FLS-1413: Update backlink text and logic

hamzabinkhalid opened the pull request:
### Ticket  
🔗 https://mhclgdigital.atlassian.net/browse/FLS-1413

---

### **Pull Request Description**

This PR addresses accessibility audit recommendations by improving sequential navigation within the Apply service forms. Previously, all form pages displayed a backlink labeled **"Go back to application overview"** (Summary page didn't show backlink), which redirected users to the task list. This disrupted the expected step-by-step navigation experience.

#### ✅ Updated Behavior:
- **First page of the form**: Backlink text remains **"Go back to application overview"**.
- **Subsequent pages (including summary)**: Backlink text is now **"Go back to previous page"**, enabling users to navigate backward through the form sequentially.

#### 🔧 Technical Improvements:
- Refactored navigation logic into a shared utility (`navigationUtils`) for consistent behavior across controllers.
- Updated `pageControllerBase` and `summaryPageController` to use the shared utility.
- Added backlink rendering logic to `summary.html` using the GOV.UK Back Link component.
- Fixed failing unit and e2e tests by updating mock requests and URLs.

---

### **Change Summary**
- Introduced shared `navigationUtils` for backlink and progress logic.
- Updated backlink text and URL logic across form pages.
- Added backlink support to the summary page.
- Fixed and updated relevant tests.

---

### **How to Test**
1. Start an application for any fund in the Apply service.
2. Open a form and observe the backlink behavior:
   - ✅ On the **first page**, the backlink should say **"Go back to application overview"**.
   - ✅ On **subsequent pages**, including the summary page, it should say **"Go back to previous page"**.

---

### ✅ Checklist
- [x] Unit tests and other appropriate tests added or updated  
- [x] Commit messages follow good practices  

---

### 📸 Screen Recording of UI Changes  

https://github.com/user-attachments/assets/f8be2c35-a22e-4b42-82fd-952291b4b8bd

nuwan-samarasinghe commented on runner/src/server/plugins/utils/UtilHelper.ts:
```diff
@@ -1,17 +1,22 @@
+  public static getBackLinkText(type: BackLinkType, isWelsh: boolean): string {
+    switch (type) {
+      case BackLinkType.Eligibility:
+        return isWelsh ? "Yn ôl at eich ceisiadau" : "Back to your applications";
```
shall we get this from en.json and cy.json

nuwan-samarasinghe commented on runner/src/server/plugins/utils/UtilHelper.ts:
```diff
@@ -1,17 +1,22 @@
+export enum BackLinkType {
+  Eligibility,
+  PreviousPage,
+  ApplicationOverview
+}
```
Nice

nuwan-samarasinghe reviewed (dismissed):
Coding wise changes are nice before EOD I will test this locally in the mean time @hamzabinkhalid you can fix the review comments

wjrm500 reviewed (changes requested):
Thanks for this @hamzabinkhalid, code looks clean and like Nuwan I'm partial to the enum 😸

However, on the "Summary" page I noticed we still don't show any "Go back to previous page". I guess the accessibility auditors meant for the back button to also be added to this page? Note in the acceptance criteria for the ticket, it says "For **every page** after this step, change this to ‘go back to previous page’ button."

hamzabinkhalid commented:
> Thanks for this @hamzabinkhalid, code looks clean and like Nuwan I'm partial to the enum 😸
> 
> However, on the "Summary" page I notice we still don't show any "Go back to previous page". I guess the accessibility auditors meant for the back button to also be added to this page? Although I agree it is not 100% clear. What would be a good reason **not** to add it to the summary page?

Thanks for flagging, I wasn't sure about this, looking into it right now. 🥲

hamzabinkhalid commented:
> Thanks for this @hamzabinkhalid, code looks clean and like Nuwan I'm partial to the enum 😸
> 
> However, on the "Summary" page I notice we still don't show any "Go back to previous page". I guess the accessibility auditors meant for the back button to also be added to this page? Although I agree it is not 100% clear. What would be a good reason **not** to add it to the summary page?

@wjrm500 Addressed this, thanks for flagging 🙂

hamzabinkhalid commented:
> Hey @hamzabinkhalid, just tested this locally and spotted a couple of issues...
> 
> * On the first page of a form (when it says "Go back to application overview"), if the form errors, the back link is not visible on the error page, see screenshot below:
> 
> <img alt="Image" width="660" height="537" src="https://private-user-images.githubusercontent.com/48405095/480017568-cf566029-e44d-4edb-9d94-9644a25f261a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3NzA5MTMsIm5iZiI6MTc1NTc3MDYxMywicGF0aCI6Ii80ODQwNTA5NS80ODAwMTc1NjgtY2Y1NjYwMjktZTQ0ZC00ZWRiLTlkOTQtOTY0NGEyNWYyNjFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIxVDEwMDMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwMWMyYTc2ZmIxYzZjMWE3ZGEyYTUzMjlhZWUwMjQ0ZTM1YmQyMmZkMjEzMzE2MDVhYTc4NmJmYWZmN2Q4MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pBsd7quTEYXgugVXB2HuAc0swRVeeXdD4d-zy-NNAyE">

@wjrm500 
Great catch, Will! I’ve identified that this issue only occurs with CTDF that has forms missing a first page. These are typically read-only, so no validation errors are expected there. That explains why the error was happening even before my changes. Either way, I’ve now addressed it and the fix is in place.

> * On any later page of a form (when it says "Go back to previous page"), if the form errors, the back link reads "Go back to application overview", yet on click the behaviour just takes you back to the previous page, see screenshots below:
> 
> <img alt="Image" width="653" height="518" src="https://private-user-images.githubusercontent.com/48405095/480017928-1e04cd82-f4d0-4158-bf86-5cdff2f0dc7e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3NzA5MTMsIm5iZiI6MTc1NTc3MDYxMywicGF0aCI6Ii80ODQwNTA5NS80ODAwMTc5MjgtMWUwNGNkODItZjRkMC00MTU4LWJmODYtNWNkZmYyZjBkYzdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIxVDEwMDMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkM2EwMmM4MDIwZDdiNTJkMzFmYzg4OTQwNmZiY2U3N2FkYjhiY2I4MGM4ZTcyODY0NTc0NTIwYTA1Y2QxY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EkyK2K7o5o3_HU6KfkgRpGCVl1rP1MqlwnNDchV7u2g"> <img alt="Image" width="659" height="735" src="https://private-user-images.githubusercontent.com/48405095/480018004-c16843e7-2046-4194-8965-0dfafbe9f420.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3NzA5MTMsIm5iZiI6MTc1NTc3MDYxMywicGF0aCI6Ii80ODQwNTA5NS80ODAwMTgwMDQtYzE2ODQzZTctMjA0Ni00MTk0LTg5NjUtMGRmYWZiZTlmNDIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIxVDEwMDMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1OTM5MDhlYmMyYmRlZDkxMTFlMWIyMjNjZjJjNzA2YmUzODkxNWZiOGE4MjY5M2U4NDZkOTVkMTBhYzY4MGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.luLL10oNpXZqWQE8TkaCyyvwJmFQDvyH7rMCQlUU_rs">

Great catch, fixed now.

> * Now that we've added a back link to the summary page, the main content has been pushed down so that the title "Summary" no longer sits at the same level on the page as the titles of question pages, which is noticeable when you switch from a question page to the summary page. We should move the main content up on the summary page to account for the inclusion of the back link, reduce the size of the gap between the back link and the title, and put the title back on the same level as in question pages. See screenshots:
> 
> <img alt="Image" width="657" height="519" src="https://private-user-images.githubusercontent.com/48405095/480019682-0556508a-e87c-47ab-bf4e-0528b5bacb71.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3NzA5MTMsIm5iZiI6MTc1NTc3MDYxMywicGF0aCI6Ii80ODQwNTA5NS80ODAwMTk2ODItMDU1NjUwOGEtZTg3Yy00N2FiLWJmNGUtMDUyOGI1YmFjYjcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIxVDEwMDMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3NmYyYzQzNTQ2OTAyZTZlMGQzMjI3NDY2OGQzNTg4MDI5YzQ4NzBkZmUyYzMyMDhiMjExNTYzZDI5OTRlMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.53Ob6Z0v5QjWoWr-IO6W3UoiNI3aE80AaM-nO4VWdPE"> <img alt="Image" width="984" height="717" src="https://private-user-images.githubusercontent.com/48405095/480019777-480e3821-b1bc-408c-82da-c2ebaf13c17c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU3NzA5MTMsIm5iZiI6MTc1NTc3MDYxMywicGF0aCI6Ii80ODQwNTA5NS80ODAwMTk3NzctNDgwZTM4MjEtYjFiYy00MDhjLTgyZGEtYzJlYmFmMTNjMTdjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIxVDEwMDMzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4ZGU2YjA5ZGY4MTY5NDg1Zjg0YzdhYTJhYTVlN2ZkNTA0OTZhMzlmOWQ0OTdlMGI0MTUwNzZmMmRlYmFiM2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JTkEouc-GwpZKnaJXHzmc4982gZ319pnq1KLI18HG0c">

Removed extra padding to make it consistent with previous pages.

> Will review the code in more depth once these issues have been ironed out, cheers

Video demo going through highlighted flows:

https://github.com/user-attachments/assets/30ee5bfb-f196-40e7-82c2-6d5dd82c8ca6

hamzabinkhalid commented:
> @hamzabinkhalid tested the app locally following looks good to me
> 
> 1. Apply process working with runner correctly and looks good
> 
> **Issues Identified**
> 
> 1. When preview the form using runner I can see some pages has the back link but not others better to have the back link over there but check with PO
> 
> <img alt="image" width="1899" height="679" src="https://private-user-images.githubusercontent.com/15814084/480548609-213a8c3c-55fd-4b58-a65c-24f8f0f479a0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyMjQwMTIsIm5iZiI6MTc1NjIyMzcxMiwicGF0aCI6Ii8xNTgxNDA4NC80ODA1NDg2MDktMjEzYThjM2MtNTVmZC00YjU4LWE2NWMtMjRmOGYwZjQ3OWEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI2VDE1NTUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1NWUxYTA0ZDVmMThiMjgwMTBkZmY5YzNkMTQ2NDhiMmU3MmQ1MGYzYjFiMmJjYzc2MDdjMGFjMTg3YWMxYjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fGg5MWllfLw9VFRAPcC-zjZHIELSQn6jvCxe-vaZS50"> <img alt="image" width="1899" height="779" src="https://private-user-images.githubusercontent.com/15814084/480548718-ad8a3bbd-5156-4fd3-8aaa-3f6e5037a3bb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyMjQwMTIsIm5iZiI6MTc1NjIyMzcxMiwicGF0aCI6Ii8xNTgxNDA4NC80ODA1NDg3MTgtYWQ4YTNiYmQtNTE1Ni00ZmQzLThhYWEtM2Y2ZTUwMzdhM2JiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI2VDE1NTUxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhMDZiYmJkMTI0MGI2ZTIxYjc1MTdkYjY3ZmIxOGJlZDgzOGMxZTcyMWFkOTdiNGMzZjQ5NTNjMjJkODY3NzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QBJKVCPrJ9CVIEyxIvdxuobY5F_KqCgwixz4CjaSAG4">
> Make sure you are testing these with FAB also
> 
> 2. Can you make the commit squash according to the changes brake them into meaning full commits if possible
> 3. Make a new e2e or unit test to test this back link behaviour

Thanks @nuwan-samarasinghe for flagging this,

Fixed the problem in preview mode, backlinks displaying on summary page but not on other pages.

Have updated the e2e repo to verify these changes in [PR](https://github.com/communitiesuk/funding-service-design-e2e-checks/pull/1069)

Video demo showing form walkthrough in preview mode from FAB:


https://github.com/user-attachments/assets/1cd602b4-8fa2-49b1-8bb5-3dcd9730fceb
